Steps4Impact/LoginV2/LoginV2ViewController.swift

```diff
@@ -0,0 +1,216 @@
+                                            alignment: .center)
+  
+  override func commonInit() {
+    super.commonInit()
+    
+    [logoImageView, onboardingImageView]
+      .forEach { $0.contentMode = .scaleAspectFit }
+    
+    logoImageView.isUserInteractionEnabled = true
+    logoImageView.addGestureRecognizer(UILongPressGestureRecognizer(
+      target: self,
+      action: #selector(logoLongPressed)))
+    
+    signInLabel.text = Strings.Login.title
+    termsAndConditionsButton.title = Strings.Login.conditions
+    termsAndConditionsButton.addTarget(self,
+                                       action: #selector(viewTermsAndConditions),
+                                       for: .touchUpInside)
+    
+    facebookLoginButton.addTarget(self,
+                                  action: #selector(facebookLoginButtonTapped),
+                                  for: .touchUpInside)
+    
+    // TODO(samisuteria): Add google and apple in a future PR
+    googleLoginButton.isHidden = true
+    appleLoginButton.isHidden = true
+  }
+  
+  override func configureView() {
+    super.configureView()
+    
+    logoImageView.snp.makeConstraints {
+      $0.height.width.equalTo(Style.Size.s128)
+    }
+    
+    stackView.addArrangedSubviews(
+      logoImageView,
+      onboardingImageView,
+      signInLabel,
+      buttonStackView,
+      termsAndConditionsButton
+    )
+    
+    buttonStackView.addArrangedSubviews(
+      facebookLoginButton,
+      googleLoginButton,
+      appleLoginButton
+    )
+    
+    view.addSubview(stackView) {
+      $0.top.equalTo(view.safeAreaLayoutGuide).inset(Style.Padding.p32)
+      $0.leading.trailing.bottom.equalToSuperview().inset(Style.Padding.p32)
+    }
+  }
+  
+  // MARK: - Actions
+  
+  @objc
+  private func logoLongPressed() {
+    let alert = AlertViewController()
+    alert.title = "Switch Server"
+    alert.body = "Switch to \(UserInfo.isStaging ? "Production" : "Staging")?"
+    alert.add(.okay({
+      UserInfo.isStaging.toggle()
+    }))
+    alert.add(.cancel())
+    AppController.shared.present(alert: alert, in: self, completion: nil)
+  }
+  
+  @objc
+  private func viewTermsAndConditions() {
+    let safariViewController = SFSafariViewController(
+      url: URL(string: "https://www.akfusa.org/website-private-policy")!)
+    present(safariViewController, animated: true, completion: nil)
+  }
+}
+
+// AKF-WCBackend
+extension LoginV2ViewController {
+  private func linkAKFBackend(fbid: String) {
+    AKFCausesService.createParticipant(fbid: fbid) { [weak self] (_) in
+      onMain {
+        AppController.shared.login()
+      }
+      self?.addParticipantToDefaultEventIfNeeded(fbid: fbid)
+    }
+  }
+  
+  private func addParticipantToDefaultEventIfNeeded(fbid: String) {
+    AKFCausesService.getParticipant(fbid: fbid) { (result) in
+      if let participant = Participant(json: result.response), participant.currentEvent == nil {
```

Comment:
```suggestion
      guard let participant = Participant(json: result.response) else { return }
      guard participant.currentEvent == nil else { return }
```

Reply:
Using guards a bit more heavily would allow you to remove the indentation.

Reply:
good call